iis-wf/iis-wf-referenceextraction/src/test/resources/eu/dnetlib/iis/wf/referenceextraction/patent/data/report_patent_metadata_extraction.json

```diff
@@ -1,2 +1,2 @@
-{"key":"processing.referenceExtraction.patent.metadataextraction.processed.total", "type":"COUNTER", "value": "7"}
-{"key":"processing.referenceExtraction.patent.metadataextraction.processed.fault", "type":"COUNTER", "value": "6"}
\ No newline at end of file
+{"key":"processing.referenceExtraction.patent.metadataextraction.processed.total", "type":"COUNTER", "value": "1"}
+{"key":"processing.referenceExtraction.patent.metadataextraction.processed.fault", "type":"COUNTER", "value": "0"}
```

Comment:
Does it mean the change related only to removing empty entries in cache and output also affects the number of faults generated by patent metadata extraction process? 

I mean it is unexpected, to me, but if it is justifiable then let's leave it as it is.

Reply:
Yes, the number of faults generated by extractor drops because we do not propagate faults from retriever as empty records to extractor. 

Before removing empty records from output we had 7 input records to retriever for which we had 7 records as output (1 non-empty and 6 empty) and 6 records as faults. The output was propagated as input to extractor. So extractor processed 7 input records with only 1 non-empty. Out of those 7 records only 1 was successfully processed and this was reflected in the report. 

After removing empty records from output we have 7 input records to retriever for which we have 1 record as output and 6 records as faults. The one output record is propagated as input to extractor which successfully processes it. And this is reflected in the new values in extractor report.